test/e2e/performanceprofile/functests/utils/nodepools/nodepools.go

```diff
@@ -67,14 +67,16 @@ func GetByClusterName(ctx context.Context, c client.Client, hostedClusterName st
 
 // AttachTuningObject is attaches a tuning object into the nodepool associated with the hosted-cluster
 // The function is idempotent
-func AttachTuningObject(ctx context.Context, cli client.Client, object client.Object) error {
-	hostedClusterName, err := hypershift.GetHostedClusterName()
-	if err != nil {
-		return err
-	}
-	np, err := GetByClusterName(ctx, cli, hostedClusterName)
-	if err != nil {
-		return err
+func AttachTuningObject(ctx context.Context, cli client.Client, object client.Object, nodePools ...*hypershiftv1beta1.NodePool) error {
+	var np *hypershiftv1beta1.NodePool
+	var err error
+	if len(nodePools) > 0 && nodePools[0] != nil {
+		np = nodePools[0]
```

Comment:
why do we need multiple nodePools passed if the function will process only the first one? I think we either pass one nodePool object and work with it, or we pass multiple and process everything. The question remains if an object can be attached to multiple nodePools or just a single one, I'm not an expert in this area though

Reply:
The goal here was to avoid affecting references in other test suites. Before this change, all the profile attachment statements for the nodepool didn’t specify any particular node pool, assuming there was only one default node pool in the environment. Now, the function will handle the first nodepool by default, while I explicitly pass the second node pool under this parameter.

Reply:
Added a separate method to remove passing of a slice unnecessary. Thanks